### ISSUE TITLE
Issue with Time out 408 on connector

### ISSUE THREAD

**Issue by iwkse** [cetra3/etherpad-alfresco#8]
Hi,
I'm having this issue since a while, the connector gives a timeout. Does that mean the APIKEY is wrong?

```
2015-10-26 00:46:18,999  INFO  [webscripts.connector.RemoteClient] [http-bio-8081-exec-12] Exception calling (GET) http://127.0.0.1:9001/socket.io/?EIO=3&transport=websocket&sid=QKcQncV5vNDYEoBXAAAF
 2015-10-26 00:46:19,000  INFO  [webscripts.connector.RemoteClient] [http-bio-8081-exec-12] Error status 408 Read timed out

WebSocket connection to 'ws://alfdev:8081/share/proxy/etherpad/socket.io/?EIO=3&transport=websocket&sid=QKcQncV5vNDYEoBXAAAF' failed: Error during WebSocket handshake: Unexpected response code: 408
```

**Comment by cetra3** [comment]
I don't believe so, it may be that it's not handling WebSocket connections correctly.

Does it still work with this error message? As it should fall back to long-polling

**Comment by iwkse** [comment]
Yes I see several xhr requests with long time, POST and GET. When I use it as the default it seems to work, but I'm trying something else using the API to clear the content of a PAD but such code is not executed at all and the debug lines I've put aren't either, when they should. So I was wondering if it's related to that issue.

**Comment by iwkse** [comment]
Not related..I found out the issue. But WebSocket fails also for you?

**Comment by cetra3** [comment]
Yeah, happy to leave this one open until it finds resulotion.  The problem is that websockets does not work through the proxy.

The answer may be to allow an alternative configuration, where clients access etherpad directly, rather than via the share proxy.

**Comment by iwkse** [comment]
Yes, this sounds interesting. Just a question about Etherpad. Do you think it can be (easily) possible to embed it without taking advantage of the IP:port setup and embed it directly, like for example it is done for tinymce or other similar editors? To me this would be a better solution.

**Comment by cetra3** [comment]
Would be great if it could be embedded, but since it runs on NodeJS this is probably not a feasible option without a complete rewrite.
